task1/FridgeManager.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace task1
+{
+    public class FridgeManager
+    {
+        public Fridge Fridge { get; set; }
+
+        public FridgeManager(Fridge fridge)
+        {
+            this.Fridge = fridge;
+            this.Fridge.FridgeStateChangedEvent += ChangingMessage;
+        }
+
+        public void ChangingMessage(object sender, FridgeHandlerEventArgs args)
+        {
+            Console.WriteLine(args.ActionMessage);
```

Comment:
Since you store state and action in EventArgs, why don't you use it here?
But you use ActionMessage only.

Reply:
-1 to logic.